src/Raven.Server/Documents/DocumentPutAction.cs

```diff
@@ -301,11 +316,31 @@ private string BuildDocumentId(string id, long newEtag, out bool knownNewId)
                         i += 19;
                         valuePtr[i] = '-';
 
-                        Format.Backwards.WriteNumber(valuePtr + i - 1, (ulong)newEtag);
+                        if (checkIfGeneratedIdIsNotOverlapping == false)
+                            Format.Backwards.WriteNumber(valuePtr + i - 1, (ulong)newEtag);
+                        else
+                        {
+                            while (true)
```

Comment:
I don't want this solution, this does a scan, and can get into bad situation.
Just change the generated doc id in this case.